feature-libs/product-configurator/rulebased/core/state/effects/configurator-basic-effect.service.spec.ts

```diff
@@ -0,0 +1,160 @@
+import { Type } from '@angular/core';
+import { TestBed, waitForAsync } from '@angular/core/testing';
+import { ConfiguratorModelUtils } from '@spartacus/product-configurator/common';
+import {
+  ATTRIBUTE_1_CHECKBOX,
+  CONFIG_ID,
+  GROUP_ID_1,
+  GROUP_ID_2,
+  GROUP_ID_3,
+  GROUP_ID_4,
+  GROUP_ID_5,
+  GROUP_ID_6,
+  GROUP_ID_7,
+  PRODUCT_CODE,
+} from '../../../testing/configurator-test-data';
+import { ConfiguratorTestUtils } from '../../../testing/configurator-test-utils';
+import { Configurator } from '../../model/configurator.model';
+import { ConfiguratorBasicEffectService } from './configurator-basic-effect.service';
+
+const group: Configurator.Group = {
+  id: GROUP_ID_1,
+  attributes: [{ name: ATTRIBUTE_1_CHECKBOX }],
+  subGroups: [],
+};
+
+const groupWithSubGroup: Configurator.Group = {
+  id: GROUP_ID_1,
+  attributes: [
+    {
+      name: ATTRIBUTE_1_CHECKBOX,
+      values: [{ name: 'val', valueCode: '1' }],
+    },
+  ],
+  subGroups: [group],
+};
+const productConfiguration: Configurator.Configuration = {
+  ...ConfiguratorTestUtils.createConfiguration(
+    CONFIG_ID,
+    ConfiguratorModelUtils.createInitialOwner()
+  ),
+  productCode: PRODUCT_CODE,
+  groups: [group, groupWithSubGroup],
+  flatGroups: [group],
+  priceSummary: {},
+  priceSupplements: [],
+};
+describe('ConfiguratorBasicEffectService', () => {
+  let classUnderTest: ConfiguratorBasicEffectService;
+
+  beforeEach(
+    waitForAsync(() => {
+      TestBed.configureTestingModule({
+        providers: [ConfiguratorBasicEffectService],
+      }).compileComponents();
+    })
+  );
+  beforeEach(() => {
+    classUnderTest = TestBed.inject(
+      ConfiguratorBasicEffectService as Type<ConfiguratorBasicEffectService>
+    );
+  });
+
+  describe('getGroupWithAttributesForConfiguration', () => {
+    it('should find group in single level config', () => {
+      expect(
+        classUnderTest.getGroupWithAttributesForConfiguration(
+          productConfiguration
+        )
+      ).toBe(GROUP_ID_1);
+    });
+    it('should throw error in case configuration has no attribute at all', () => {
+      expect(function () {
+        classUnderTest.getGroupWithAttributesForConfiguration(
+          ConfiguratorTestUtils.createConfiguration(
+            'a',
+            ConfiguratorModelUtils.createInitialOwner()
+          )
+        );
+      }).toThrow();
+    });
+  });
+
+  describe('getGroupWithAttributes', () => {
+    it('should find group in multi level config', () => {
+      const groups: Configurator.Group[] = [
+        {
+          id: GROUP_ID_1,
```

Comment:
Please, change group IDs because it is misleading why the group that contains any attributes has a `GROUP_ID_1` ID as well

Reply:
Agree